packages/fx-core/tests/component/generator/generator.test.ts

```diff
@@ -539,6 +539,25 @@ describe("Generator happy path", async () => {
     assert.isTrue(success);
     mockedEnvRestore();
   });
+
+  it("template from fallback", async () => {
+    const foobarTemplateName = "foobarTemplate";
+    const realTemplateName = "non-sso-tab";
+    const language = "ts";
+    const inputDir = path.join(tmpDir, "input");
+    await fs.ensureDir(path.join(inputDir, foobarTemplateName));
+    const fileData = "{{appName}}";
+    await fs.writeFile(path.join(inputDir, foobarTemplateName, "test.txt.tpl"), fileData);
```

Comment:
Seems you just want to create an AdmZip object. Is it necessary to write the file to the file system?

Reply:
Thanks for the feedback, I'll add a success check for generating real template and a check to make sure no foobar files are generated.

Reply:
Yes, you're right, because foobarTemplateName is only needed to make FilterFn unable to pass the prefix checksum, no need to add the contents of the file.

Reply:
I'll remove the code for adding file content.